### DISCUSSION TITLE
[DEPRECATED] Informal changelog for telemetry output

### DISCUSSION THREAD

**Discussion by markfields** [microsoft/FluidFramework#9014] [Contributors]
As discussed in #6976, **the shape of the telemetry output coming from FF code is not stable and is unfit for programmatic consumption**.  That said, operationally our team uses this data for diagnostic investigations and it's inconvenient/confusing when it changes.

I'd like to try using this discussion as a place to log changes to telemetry output, e.g. event names or property names.  This will give us a single place to check when we can't find what we're looking for or want to educate ourselves on any recent changes. Obviously, this will not be a comprehensive list since it's opt-in, but the more it's used the more useful it will become.

## How-To

1. After merging a PR that changes telemetry output in a way that someone might want to know about, reply to this discussion with a link to the PR, and include the old/new values.
2. Searching via GitHub's universal `/` search box for a property or event name and filtering to Discussions will yield results from any posts below.
3. Consider subscribing to this issue to get notifications when someone posts here.

**Comment by markfields** [comment]
https://github.com/microsoft/FluidFramework/pull/9016 updated naming of props on some DataCorruptionErrors (`errorType` === `dataCorruptionError`) to be consistent with other places logging the same data

### Old names

```
    messageClientId (unchanged)
    sequenceNumber
    clientSequenceNumber
    referenceSequenceNumber
    minimumSequenceNumber
    messageTimestamp (unchanged)
```

### New names

```
    messageClientId
    messageSequenceNumber
    messageClientSequenceNumber
    messageReferenceSequenceNumber
    messageMinimumSequenceNumber
    messageTimestamp
```

**Comment by markfields** [comment]
#8449 and #9045 renamed the telemetry prop `X-RequestStats` to `XRequestStatsHeader`.

Note: In Office Loop telemetry internal to Microsoft, the old column name was `Data_X_RequestStats` (replacing `-` with `_`)

**Comment by markfields** [comment]
#9021 made some changes to Summary telemetry, here's what's called out in the PR description:

1. Added `referenceSequenceNumber` to fluid:telemetry:OdspDriver:uploadSummary_end event
2. changed "GenerateSummary" event to call `PerformanceEvent` .
3.  removing `uploadSummaryWithContext` event in odsp
3. adding fullTree, timeSinceLastAttempt, timeSinceLastSummary to `GenerateSummary` event
4. renamed "GenerateSummary" event to `Summarize_generate`, renamed "SummaryOp" to `Summarize_Op`
5. on summarize_end event, rename  sequenceNumber to `ackNackSequenceNumber`, added referenceSequenceNumber
6. using one variable `generateTelemetryProps` for both `summarize_end` and `summarize_generate` events#9140 

@chensixx - feel free to edit this insofar as it would be helpful to add additional details or correct anything.

**Comment by markfields** [comment]
#9249 changes many error messages for errors raised from the FF code:

Old messages (or key substrings) that have been updated or replaced with friendlier sentences:

```
ExistingContextDoesNotSatisfyIncomingProposal
clientClosingConnection
Invalid filename
OdspSocketError
odspFetchError 
No response from fetch call
OdspFetchOffline
Fetch error
errorWhileParsingFetchResponse
Token is null for ${name} call
storageServiceDisposedCannotRetry
R11sSocketError
r11sFetchError
containerClosedWithoutErrorDuringLoad
containerNotValidForAttach 
existingContextDoesNotSatisfyIncomingProposal
messageClientIdMissingFromQuorum
messageClientIdShouldHaveLeft
twoMessagesWithSameSeqNumAndDifferentPayload
msnMovesBackwards
nonSequentialSequenceNumber
failedToRetrieveOpsFromStorage:TooManyRetries
SummaryMetadataMismatch
MaxReconnectsWithNoProgress
orderSequentiallyCallbackException
duplicateDataStoreCreatedWithExistingId
summaryAckWaitTimeout
summaryNack: ${message}
unableToGetAccessToken
unableToGetChildren
unableToGetDriveItemIdFromPath
failedToCreateFile
unableToGetDriveItemIdFromPath
failedToGetDriveResponse
```

Additionally, these words were present in error messages and have been updated to match the spelling of the handler names they represent:

```
connectError
connectTimeout
connectDocumentSuccess
connectDocumentError
```

**Comment by markfields** [comment]
#9306 changed many of the performance properties logged for the "TreesLatest" event.
The change renamed some fields and removed 2 others: 
- secureConntime -> secureConnectionTime
- responsenetworktime -> responseNetworkTime
- tcphandshakeTime -> tcpHandshakeTime
- fetchstarttorespendtime -> fetchStartToResponseEndTime
- reqstarttorespendtime -> reqStartToResponseEndTime 
- overalltime was removed (equivalent to duration)
- clienttime was removed (it can be calculated by duration - networkTime)

**Comment by NicholasCouri** [comment]
Once we fix #9248 
GetSharingLinkToken  will now be split in 2 different entries:  GetFileLinkCore_GetToken and  GetFileItemLite_GetToken
